File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.py
def solution(arr, queries):
    answer = []
    for s,e,k in queries:
        result = []
        for key,n in enumerate(arr):
            if s <= key and key <= e and k < n:
                result.append(n)
        answer.append(min(result) if len(result) > 0 else -1)
    return answer

File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr, int[][] queries) {
        int[] answer = new int[queries.length];
        
        int i = 0;
        
        for (int[] query : queries) {
            int[] array = Arrays.copyOfRange(arr, query[0], query[1] + 1);
            Arrays.sort(array);
            
            int ans = -1;
            for (int num : array) {
                if (query[2] < num) {
                    ans = num;
                    break;
                }
            }
            
            answer[i++] = ans;
        }
        return answer;
    }
}

